libraries/AP_Scripting/examples/rev_thr_on_land.lua

```diff
@@ -0,0 +1,181 @@
+local RC_ARM_CHANNEL_                = bind_add_param('ARM_CHANNEL',              2, 6)
+local ARM_CHANNEL_HI_PWM_            = bind_add_param('ARM_HI_PWM',               3, 1500)
+
+--REV activation and spool up parameters
+local LAND_DET_TIME_                 = bind_add_param('LAND_TIME',                4, 1)
+local THRESH_LAND_HEIGHT_            = bind_add_param('LAND_HEIGHT',              5, 0.12)
+local TIME_THROT_WAIT_MS_            = bind_add_param('WAIT_MS',                  6, 1000)
+local REV_THRUST_SPOOL_DPWM_         = bind_add_param('SPOOL_DPWM',               7, 60)
+local REV_THRUST_CHG_PWM_PS_         = bind_add_param('CHGPWM_PS',                8, 150)
+
+--REV strength and duration setup
+local REV_THRUST_IDLE_PWM_           = bind_add_param('THR_IDLE_PWM',             9, 990)
+local REV_THRUST_HIGH_PWM_           =  bind_add_param('HI_PWM',                 10, 1600)
+local MAXTIME_REV_THRUST_APPLIED_MS_ = bind_add_param('MAXTIME_MS',              11, 10000)
+local MIN_GROUND_SPEED_              = bind_add_param('GNDSPEED_MIN',            12, 2.5)
+
+-- *****---INTERNAL VARIABLES AND BOOL FLAGS --- DO NOT CHANGE ANYTHING HERE UNLESS MODIFYING THE CODE---*****
+local AP_VERSION_NUM_ = 10 * tonumber(FWVersion:major()) + tonumber(FWVersion:minor())
+local RFND_ROT_DN_ = 25
+local MAX_DIST_ = rangefinder:max_distance_cm_orient(RFND_ROT_DN_) * 0.01
+local MODE_FBWA = 5
+local COUNT_IN_RANGE_ = 0
+local RCMAP_THROTTLE_  = bind_param("RCMAP_THROTTLE")
+local THROT_CHANNEL_ = rc:get_channel(RCMAP_THROTTLE_:get())
+local REV_THRUST_AUX_SW_ = rc:find_channel_for_option(64)
+local REV_THRUST_SW_LOW_PWM_ = 1000
+local REV_THRUST_SW_HIGH_PWM_ = 2000
+local TIME_SEC_ = 0
+local CALLBACK_TIME_ = 200
+local IS_THR_SUP_ = true
+local IS_IN_REV_THRUST_ = false
+local REV_THRUST_SW_ = false
+
+local REV_THRUST_CUR_PWM_ = REV_THRUST_IDLE_PWM_:get() + REV_THRUST_SPOOL_DPWM_:get()
+local TIM_AFTER_TOUCHDN_ = LAND_DET_TIME_:get() * 1000 * 0.01
+
+function UPDATE()
+  if (REV_ENABLE_:get() < 1) then
+    return
+  end
+  if (vehicle:get_mode() == MODE_FBWA and (arming:is_armed())
+      and (rc:get_pwm(RC_ARM_CHANNEL_:get()) > ARM_CHANNEL_HI_PWM_:get())
+      and (THROT_CHANNEL_:norm_input_dz() == 0)) then
+    if (RFND_READING() < THRESH_LAND_HEIGHT_:get()) then
+      if (COUNT_IN_RANGE_ > TIM_AFTER_TOUCHDN_) then
+        TIME_SEC_ = millis():tofloat()
+        return ENGAGE_REV_THRUST()
+      end
+      COUNT_IN_RANGE_ = COUNT_IN_RANGE_ + 1
+    else
+      COUNT_IN_RANGE_ = 0
+    end
+  end
+  return UPDATE, 100 -- check again in 10Hz
+end
+
+function RFND_READING()
+  if rangefinder:has_data_orient(RFND_ROT_DN_) then
+    return rangefinder:distance_cm_orient(RFND_ROT_DN_) * 0.01
+  end
+  return MAX_DIST_
+end
+
+function DISENGAGE_REV_THRUST()
+  THROT_CHANNEL_:set_override(REV_THRUST_IDLE_PWM_:get())
+  REV_THRUST_AUX_SW_:set_override(REV_THRUST_SW_LOW_PWM_)
```

Comment:
Rather than these overrides you can trigger the aux function directly. Although I think you will still need to have a switch setup due to the way the init is done....

```suggestion
    rc:run_aux_function(64, 0)
```